File: message/message-cs/src/quan/message/test/RoleInfo.cs
using System;

namespace quan.message.test
{
    public class RoleInfo : Bean
    {
        public long roleId { get; set; } = 0L; //角色id

        private string _roleName = ""; //角色名

        public string roleName
        {
            get => _roleName;
            set => _roleName = value ?? throw new NullReferenceException();
        }


        public override void encode(Buffer buffer)
        {
            base.encode(buffer);
            buffer.WriteLong(roleId);
            buffer.WriteString(roleName);
        }

        public override void decode(Buffer buffer)
        {
            base.decode(buffer);
            roleId = buffer.ReadLong();
            roleName = buffer.ReadString();
        }

        public override string ToString()
        {
            return $"{nameof(roleId)}: {roleId}, {nameof(roleName)}: {roleName}";
        }
    }
}

File: quan.java/message/src/test/java/quan/message/user/RoleInfo.java
package quan.message.user;

import java.util.*;
import quan.message.*;

/**
 * 角色信息2<br/>
 * 代码自动生成，请勿手动修改
 */
public class RoleInfo extends Bean {

    //角色id
    private long id;

    //角色名
    private String name = "";

    public RoleInfo() {
    }
    
    public RoleInfo(long id, String name) {
        this.setId(id);
        this.setName(name);
    }

    /**
     * 角色id
     */
    public long getId() {
        return id;
    }

    /**
     * 角色id
     */
    public RoleInfo setId(long id) {
        this.id = id;
        return this;
    }

    /**
     * 角色名
     */
    public String getName() {
        return name;
    }

    /**
     * 角色名
     */
    public RoleInfo setName(String name) {
        Objects.requireNonNull(name);
        this.name = name;
        return this;
    }

    @Override
    public void encode(Buffer buffer) {
        super.encode(buffer);

        buffer.writeLong(this.id);
        buffer.writeString(this.name);
    }

    @Override
    public void decode(Buffer buffer) {
        super.decode(buffer);

        this.id = buffer.readLong();
        this.name = buffer.readString();
    }

    @Override
    public String toString() {
        return "RoleInfo{" +
                "id=" + id +
                ",name='" + name + '\'' +
                '}';

    }

}
